### DISCUSSION TITLE
SIP 4 - Change operator set

### DISCUSSION THREAD

**Discussion by alonmuroch** [ssvlabs/SIPs#14] [SIP Discussions]
merged, open for discussion

**Comment by alonmuroch** [comment]
@lior-blox there is an issue with the current spec, nothing guarantees that the old committee will maintain the storage + subscribed to the topic to serve the highest decided to the new committee.
We might need to limit how many operators you can change, for example no more than f operators. This will make sure there are 2f+1 operators at least with the decided data
WYT?

**Reply by liorrutenberg** [reply]
It still not ensure it, no?
for example if you have more than f faulty/inactive, and it will require the user to make multiple update transactions.

**Reply by alonmuroch** [reply]
If you have >f faulty you need to change your operator set completely as it might be compromised. 
That will mean deactivate the validator on SSV or even potentially exit it

**Comment by alonmuroch** [comment]
@lior-blox @nivBlox @amir-blox change operator has impact on syncing.
If the operator set can change then it means a share has versions which also means that when we accept decided msgs (or any QBFT msgs) it has to be in reference to a specific version of a share + how do we know which version is the relevant one for a specific epoch?

**Reply by alonmuroch** [reply]
That goes to future decided as well as past

**Comment by alonmuroch** [comment]
Notes on change operator challenges
https://hackmd.io/TBJUjofqQMmcGTIxwYOf4g
